packages/fleather/example/lib/main.dart

```diff
@@ -113,3 +141,41 @@ class _HomePageState extends State<HomePage> {
     }
   }
 }
+
+class ForceNewlineForInsertsAroundInlineImageRule extends InsertRule {
+  @override
+  Delta? apply(Delta document, int index, Object data) {
+    if (data is! String) return null;
+
+    final iter = DeltaIterator(document);
+    final previous = iter.skip(index);
+    final target = iter.next();
+    final cursorBeforeBlockEmbed = _isInlineImage(target.data);
+    final cursorAfterBlockEmbed =
+        previous != null && _isInlineImage(previous.data);
+
+    if (cursorBeforeBlockEmbed || cursorAfterBlockEmbed) {
+      final delta = Delta()..retain(index);
+      if (cursorAfterBlockEmbed && !data.startsWith('\n')) {
+        delta.insert('\n');
+      }
+      delta.insert(data);
+      if (cursorBeforeBlockEmbed && !data.endsWith('\n')) {
```

Comment:
Shouldn't this apply if cursorAfterBlockEmbed isn't not verifed?

Reply:
I am not sure what you mean by that.  

We need both conditions in case user is typing text in-between two inline images.

Reply:
With the current implementation, if both cursorBefore & cursorAfter are verified (typing text in between embeds), the rule inserts new line twice

Reply:
Yes. If user types text in-between two images we want to insert newline after an image, then insert text and finally insert a new line before second image.

The only problem in this case is that ideally, we would move cursor by one position back.

Reply:
Right, my bad